www/caddy/src/opnsense/scripts/OPNsense/Caddy/caddy_control.py

```diff
@@ -76,6 +74,15 @@ def run_service_command(action, action_message):
     if action in actions:
         service_action = actions[action]
         message = f"{action.capitalize()}ing Caddy service" if action != "validate" else "Validating Caddy configuration"
+
+        # Call setup script for 'validate' and 'reload' actions
+        # This is needed because the setup script triggers the caddy_certs.php script, which exports all certificates into the filesystem.
+        # Caddy reloads certificates when reloadssl is used. Because it is a non standard command, the caddy_setup script will not be triggered in /etc/rc.conf.d/caddy.
+        # The validate command needs it to make sure all certificates are in the filesystem, because otherwise the validation fails.
+        if action in ["validate", "reload"]:
```

Comment:
one note here is that we can't really trace `reloadssl` because it's being labeled as `reload`. it gets confusing when code is shifted again and "reload" is the normal reload that runs setup.sh automatically... :)

Reply:
I would like to map this one to one. But how can I tell the reconfigureAction() that it should call ``reloadssl`` and not ``reload``?

https://github.com/opnsense/core/blob/6709625ad69c42908f9533f6c887a3ca9161c6da/src/opnsense/mvc/app/controllers/OPNsense/Base/ApiMutableServiceControllerBase.php#L181

https://github.com/opnsense/plugins/blob/90fe67c97fcb41f0de5082cefc960ba8804647c9/www/caddy/src/opnsense/mvc/app/controllers/OPNsense/Caddy/Api/ServiceController.php#L44-L48

Reply:
it goes back to the reload vs. reloadssl... reloadssl should have been reload and reload should have been reloadnossl ;)

You can maybe change the script matching to make it a little more clear

Reply:
reverse match over this... https://github.com/opnsense/plugins/pull/3982/files/0bf65bb2fd02541065f46c72292e38eb545e2806#diff-7cf6fb27c641e99d4a974c1b644d1aede66e80ae229a269e5f3ece8a1777dbe3R68

Reply:
I'll try that. I doubt upstream will fix it if I put an issue since it's a breaking change for anybody who only wants the normal reload vs. reloadssl.